server/api_authenticate.go

```diff
@@ -217,37 +217,60 @@ func (s *ApiServer) AuthenticateEmail(ctx context.Context, in *api.AuthenticateE
 	}
 
 	email := in.Account
-	if email == nil || email.Email == "" || email.Password == "" {
+	if email == nil {
 		return nil, status.Error(codes.InvalidArgument, "Email address and password is required.")
+	}
+
+	var attemptUsernameLogin bool
+	if email.Email == "" {
+		// Password was supplied, but no email. Perhaps the user is attempting to login with username/password.
+		attemptUsernameLogin = true
 	} else if invalidCharsRegex.MatchString(email.Email) {
 		return nil, status.Error(codes.InvalidArgument, "Invalid email address, no spaces or control characters allowed.")
-	} else if len(email.Password) < 8 {
-		return nil, status.Error(codes.InvalidArgument, "Password must be longer than 8 characters.")
 	} else if !emailRegex.MatchString(email.Email) {
 		return nil, status.Error(codes.InvalidArgument, "Invalid email address format.")
 	} else if len(email.Email) < 10 || len(email.Email) > 255 {
 		return nil, status.Error(codes.InvalidArgument, "Invalid email address, must be 10-255 bytes.")
 	}
 
-	cleanEmail := strings.ToLower(email.Email)
+	if len(email.Password) < 8 {
+		return nil, status.Error(codes.InvalidArgument, "Password must be longer than 8 characters.")
```

Comment:
Must be 8 or longer

Reply:
Unfortunately the current API error response structure means clients rely on error strings to disambiguate between various bad input cases. I don't think we should change it in case it breaks client error handling